### PR TITLE
Add release workflow

### PR DESCRIPTION
**Pull request by dhaiducek**
- https://github.com/stolostron/backlog/issues/26085

### CODE REVIEW

**Comment by dhaiducek** [comment]
This is largely taken from `clusteradm`'s workflow and will generate a draft release whenever a tag is pushed. I've updated the Policy Generator release instructions accordingly: https://docs.google.com/document/d/1puzltjSbIoNiNA9uMhWLOZGyYPj69ZGIhvnxTo0ihc4/edit#

File: .github/release.yml [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+
+name: Create Release
+
+jobs:
+  release:
+    name: Create Release
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
```
Comment (mprahl) [inline .github/release.yml:16]: Optional:
```suggestion
        uses: actions/checkout@v3
```

File: .github/release.yml [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Set up Go
+        uses: actions/setup-go@v3
```
Comment (mprahl) [inline .github/release.yml:19]:
```suggestion
        uses: actions/setup-go@v3
        with:
          go-version-file: go.mod
```

File: .github/release.yml [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+      - name: Build project
+        run: |
+          make build-binary
```
Comment (mprahl) [inline .github/release.yml:23]:
```suggestion
          make build-release
```

File: .github/release.yml (hunk 1 above)
Comment (dhaiducek) [reply .github/release.yml:16]: There is no `v3` for `checkout`: https://github.com/actions/checkout/releases

File: .github/release.yml (hunk 2 above)
Comment (dhaiducek) [reply .github/release.yml:19]: Ah, missed this one--thanks!

**Comment by dhaiducek** [comment]
Sorry--the file was in the `.github` folder instead of `.github/workflows` and I moved it, updated the wrong file, and then deleted the updated file. 🤦 Should be good now. 🙂

File: .github/release.yml (hunk 1 above)
Comment (mprahl) [reply .github/release.yml]: @dhaiducek there is, but v2 is also supported still:
https://github.com/actions/checkout/releases/tag/v3.1.0

**Review by mprahl — APPROVED** [review]

**Pull request merged**